components/layout_2020/taffy/layout.rs

```diff
@@ -0,0 +1,620 @@
+                    .to_f32_px(),
+            ),
+            height: auto_or_to_option(content_box_size_override.block_size)
+                .map(Au::to_f32_px)
+                .maybe_add(pbm.padding_border_sums.block.to_f32_px()),
+        };
+
+        let taffy_containing_block = taffy::Size {
+            width: Some(containing_block.inline_size.to_f32_px()),
+            height: auto_or_to_option(containing_block.block_size).map(Au::to_f32_px),
+        };
+
+        let layout_input = taffy::LayoutInput {
+            run_mode: taffy::RunMode::PerformLayout,
+            sizing_mode: taffy::SizingMode::InherentSize,
+            axis: taffy::RequestedAxis::Vertical,
+            vertical_margins_are_collapsible: taffy::Line::FALSE,
+
+            known_dimensions,
+            parent_size: taffy_containing_block,
+            available_space: taffy_containing_block.map(taffy::AvailableSpace::from),
+        };
+
+        let output = match flex_context.style.clone_display().inside() {
+            DisplayInside::Grid => {
+                taffy::compute_grid_layout(&mut flex_context, DUMMY_NODE_ID, layout_input)
+            },
+            _ => panic!("Servo is only configured to use Taffy for CSS Grid layout"),
+        };
+
+        // Convert `taffy::Layout` into Servo `Fragment`s
+        let fragments: Vec<Fragment> = self
+            .children
+            .iter()
+            .map(|child| (**child).borrow_mut())
+            .map(|mut child| {
+                fn rect_to_logical_sides<T>(rect: taffy::Rect<T>) -> LogicalSides<T> {
+                    LogicalSides {
+                        inline_start: rect.left,
+                        inline_end: rect.right,
+                        block_start: rect.top,
+                        block_end: rect.bottom,
+                    }
+                }
+
+                fn rect_to_physical_sides<T>(rect: taffy::Rect<T>) -> PhysicalSides<T> {
+                    PhysicalSides {
+                        left: rect.left,
+                        right: rect.right,
+                        top: rect.top,
+                        bottom: rect.bottom,
+                        _unit: PhantomData,
+                    }
+                }
+
+                fn size_and_pos_to_logical_rect<T: Default>(
+                    position: taffy::Point<T>,
+                    size: taffy::Size<T>,
+                ) -> PhysicalRect<T> {
+                    PhysicalRect {
+                        origin: PhysicalPoint {
+                            x: position.x,
+                            y: position.y,
+                            _unit: PhantomData,
+                        },
+                        size: PhysicalSize {
+                            width: size.width,
+                            height: size.height,
+                            _unit: PhantomData,
+                        },
+                    }
+                }
```

Comment:
Please use the constructors `PhysicalRect::new()`, `PhysicalPoint::new()` and `PhysicalSize::new()`.